presto-main/src/main/java/io/prestosql/operator/OrderByOperator.java

```diff
@@ -409,6 +438,18 @@ private void updateMemoryUsage()
                 revocableMemoryContext.setBytes(pageIndex.getEstimatedSize().toBytes());
             }
         }
+        else if (spillEnabled && state == State.SECONDAY_INPUT) {
+            if (secondaryPageIndex.getPositionCount() == 0) {
+                secondaryRevocableMemoryContext.setBytes(0);
+            }
+            else {
+                secondaryRevocableMemoryContext.setBytes(secondaryPageIndex.getEstimatedSize().toBytes());
+                if (secondaryRevocableMemoryContext.getBytes() > secondarySpillThreshold) {
+                    getFutureValue(spillToDisk());
```

Comment:
THis will block the addInput() function.. and it will exceed its Quanta for scheduled time... this should not block here ever.